presto-testng-services/src/test/java/io/prestosql/testng/services/TestReportUnannotatedMethods.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.testng.services;
+
+import io.prestosql.tempto.Requirement;
+import io.prestosql.tempto.Requirements;
+import io.prestosql.tempto.RequirementsProvider;
+import io.prestosql.tempto.configuration.Configuration;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestReportUnannotatedMethods
+{
+    private final ReportUnannotatedMethods instance = new ReportUnannotatedMethods();
+
+    @Test
+    public void testTest()
```

Comment:
Yeah... not the best name.

Reply:
Actually, the whole class could use less "test*" words.

Reply:
Why? The problem is that it noun and verb. This test tests the test. Notice that `TestingTest` has a method annotated with `@Test`. While other test cases are different.

What do you suggest?

Reply:
> Actually, the whole class could use less "test*" words.

in fact, it's legacy from JUnit 3 era, when Java didn't have annotations and method name stood for `@Test`; new code shouldn't use such convention

there are many, many more test naming problems in Presto, subject for huuuge refactoring some day:
- `TestXxxx` (attribute + object in wrong order) - the standard way of naming this pretty everywhere is `XxxxTest` / `XxxxSpec`, etc (natural order: attributive + subject)
- `TestingXxxx` (gerund + object - too ambiguous) - the standard way of naming this is `TestXxxx` / `MockXxxx` / `StubXxxx` / `TestXxxxService` / `TestXxxxUtil` etc (natural order: attribute + object)
- `testSomethingVeryVeryVeryLongWhenSomethingLong` - hard to read, camel case is terrible for more descriptive test names; better break Java conventions and allow `something_very_very_long_when_something_long` (for this reason languages like Kotlin or Groovy allow even spaces and most special chars in identifiers!)

Reply:
> > Actually, the whole class could use less "test*" words.
> 
> in fact, it's legacy from JUnit 3 era, when Java didn't have annotations and method name stood for `@Test`; new code shouldn't use such convention

Intellij's `⌘O` does not let my find test classes by name, unless they have something in their name
similarily, in a test class with tests  helper methods (normal thing), `⌘F12` does not let me find test methods my name, unless they have something in their name

-- the naming convention is for us, developers, even more than testng, surefile or whatever

Reply:
You need to find by method names probably only because some test classes are huuuge here. 
If all test classes were small (this is another refactoring needed in some places), you didn't have to find by method names, and class name pattern like *Test is enough.

Reply:
> If all test classes were small

Then we would use "find class by name"